src/IO/S3/PocoHTTPClient.cpp

```diff
@@ -395,6 +445,19 @@ void PocoHTTPClient::makeRequestInternalImpl(
     addMetric(request, S3MetricType::Count);
     CurrentMetrics::Increment metric_increment{CurrentMetrics::S3Requests};
 
+    UInt64 connect_time = 0;
+    UInt64 first_byte_time = 0;
+    bool latency_recorded = false;
+    S3LatencyType first_byte_latency_type = S3LatencyType::FirstByteAttempt1;
+
+    if (!first_attempt)
```

Comment:
lest make a method for the logic which determine `first_byte_latency_type` value.